app/src/main/java/ch/epfl/sdp/drone3d/ui/mission/ItineraryShowActivity.kt

```diff
@@ -84,12 +96,20 @@ class ItineraryShowActivity : BaseMapActivity() {
                 }
             }
         }
+
         deleteButton = findViewById(R.id.mission_delete)
         deleteButton.visibility =
             if (authService.getCurrentSession()?.user?.uid == ownerUid) View.VISIBLE else View.GONE
 
         goToMissionInProgressButton = findViewById(R.id.buttonToMissionInProgressActivity)
         goToMissionInProgressButton.isEnabled = canMissionBeLaunched()
+
+        if (currentMissionPath != null && currentMissionPath!!.isNotEmpty()) {
+            weatherReport = weatherService.getWeatherReport(LatLng(currentMissionPath!![0].latitude, currentMissionPath!![0].longitude))
```

Comment:
Why did you get the latitude and longitude of a LatLng to create a new one instead of passing it directly? I think that you could replace this line with the following:

weatherReport = weatherService.getWeatherReport(currentMissionPath!![0])